File: src/connections/network/BroadcastManager.java
package connections.network;

import java.io.IOException;
import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.net.SocketException;
import davinci.Device;
import gui.img.ImageIcons;

public class BroadcastManager
{
    private DatagramSocket serverSocketUDP = null;

    public BroadcastManager()
        throws SocketException
    {
        serverSocketUDP = new DatagramSocket(5000);
    }

    public boolean networkIsAvailable()
    {
        // TODO
        return true;
    }

    public Device search()
    {
        final byte[] receiveData = new byte[1024];

        try
        {
            final DatagramPacket receivePacket = new DatagramPacket(
                receiveData,
                receiveData.length);
            serverSocketUDP.setSoTimeout(1000);
            serverSocketUDP.receive(receivePacket);

            final String message =
                new String(receivePacket.getData()).trim();

            return new Device(ImageIcons.getInstance().getWifiIcon(), message.split(","));

        } catch (IOException e)
        {
            return null;
        }
    }
}

File: src/connections/network/BroadcastManager.scala
package connections.network

import java.net.{DatagramPacket, DatagramSocket}

import davinci.Device
import gui.img.ImageIcons

// TODO: Don't want to reference the GUI package

class BroadcastManager extends DatagramSocket(5000) {

    this.setSoTimeout(1000)

    /**
     * Starts searching for devices on the network.
     * @return  First detected device.
     */
    def search: Device = {
        val receiveData: Array[Byte] = new Array[Byte](1024)

        try {
            val receivePacket: DatagramPacket = new DatagramPacket(receiveData, receiveData.length)

            this.receive(receivePacket)

            val message: String = new String(receivePacket.getData).trim
            new Device(ImageIcons.wifiIcon, message.split(","))
        } catch {
            case e: Exception => null
        }
    }
}
